controller/array_action/array_action_types.py

```diff
@@ -33,3 +33,13 @@ def __init__(self, host_id, host_name, iscsi_names, wwns):
         self.name = host_name
         self.iscsi_names = iscsi_names
         self.wwns = wwns
+
+
+class Replication:
```

Comment:
why not do it as a dataclass?

Reply:
not sure.
why were others changed to dataclass? why wasn't `Host` changed?

Reply:
it's a good idea to change `Host` too. 
dataclass helpful methods ( e.g. "toString" and other)..

Reply:
then sounds to me like it can be done across the project (i.e. at least for `Host` as well) in a separate PR.
we don't have automation for replication yet, so I'm not keen to make optional changes with potential breakage before GA

Reply:
one disadvantage to convert to dataclass is that it would be harder to add logic to those classes later (e.g. `__eq__`)

Reply:
are we? it's just like [any other class ](https://www.tutorialspoint.com/data-classes-in-python-dataclasses#:~:text=The%20dataclasses%20is%20a%20new%20module%20added%20in,a%20few%20more%29%20for%20a%20user%20defined%20class.)in that regard.. (any custom one will shadow the automatically generated ones)

Reply:
right, my bad, it won't be harder, just inappropriate

Reply:
why inappropriate?

Reply:
because dataclasses by their nature shouldn't contain much logic

Reply:
great, we got none

Reply:
that's why I [wrote "later"](https://github.com/IBM/ibm-block-csi-driver/pull/366#discussion_r705174108)

Reply:
Do you reckon there will be changes in the future to the type classes?

Reply:
it's hard to tell, we do need to split the big files somehow...
regardless of said disadvantage, I [don't think it's a must](https://github.com/IBM/ibm-block-csi-driver/pull/366#discussion_r705171484) for this PR

Reply:
ok, I still think its better but it's not blocker IMO